commit 78b5ea1f6bf67c87ed0b3375596187a7986d7e8a
Author: Klemens David Morgenstern <klemens.morgenstern@gmx.net>
Date:   Mon Nov 12 14:48:58 2018 +0700

    checking the sigtimedwait macro --> test should fail on OSX & work on Linux

diff --git a/include/boost/process/detail/config.hpp b/include/boost/process/detail/config.hpp
index dfea2c3..55ca081 100644
--- a/include/boost/process/detail/config.hpp
+++ b/include/boost/process/detail/config.hpp
@@ -57,6 +57,10 @@ inline std::error_code get_last_error() noexcept
 #define BOOST_POSIX_HAS_VFORK 1
 #endif
 
+#if (_POSIX_C_SOURCE >= 199309L)
+#define BOOST_POSIX_HAS_SIGTIMEDWAIT 1
+#endif
+
 #elif defined(BOOST_WINDOWS_API)
 namespace windows {namespace extensions {}}
 namespace api = windows;
diff --git a/include/boost/process/detail/posix/wait_for_exit.hpp b/include/boost/process/detail/posix/wait_for_exit.hpp
index 2e2c751..7880174 100644
--- a/include/boost/process/detail/posix/wait_for_exit.hpp
+++ b/include/boost/process/detail/posix/wait_for_exit.hpp
@@ -57,8 +57,8 @@ inline bool wait_until(
 
     ::sigset_t  sigset;
 
-    ::sigemptyset(&sigset);
-    ::sigaddset(&sigset, SIGCHLD);
+    sigemptyset(&sigset);
+    sigaddset(&sigset, SIGCHLD);
 
     auto get_timespec = 
             [](const Duration & dur)
@@ -81,6 +81,7 @@ inline bool wait_until(
 
     bool timed_out;
 
+#if defined(BOOST_POSIX_HAS_SIGTIMEDWAIT)
     do
     {
         auto ts = get_timespec(time_out - Clock::now());
@@ -101,6 +102,7 @@ inline bool wait_until(
     while ((ret == 0) ||
           (((ret == -1) && errno == EINTR) ||
            ((ret != -1) && !WIFEXITED(status) && !WIFSIGNALED(status))));
+#endif
 
     if (ret == -1)
         ec = boost::process::detail::get_last_error();
diff --git a/include/boost/process/detail/posix/wait_group.hpp b/include/boost/process/detail/posix/wait_group.hpp
index ff2fe6d..2f17156 100644
--- a/include/boost/process/detail/posix/wait_group.hpp
+++ b/include/boost/process/detail/posix/wait_group.hpp
@@ -61,8 +61,9 @@ inline bool wait_until(
     ::sigset_t  sigset;
     ::siginfo_t siginfo;
 
-    ::sigemptyset(&sigset);
-    ::sigaddset(&sigset, SIGCHLD);
+    sigemptyset(&sigset);
+    sigaddset(&sigset, SIGCHLD);
+
 
     auto get_timespec = 
             [](const Duration & dur)
@@ -84,6 +85,7 @@ inline bool wait_until(
         return false;
     }
 
+#if defined(BOOST_POSIX_HAS_SIGTIMEDWAIT)
     do
     {
         auto ts = get_timespec(time_out - Clock::now());
@@ -99,13 +101,13 @@ inline bool wait_until(
             return false; 
         }
 
-
         //check if we're done
         ret = ::waitid(P_PGID, p.grp, &siginfo, WEXITED | WNOHANG);
 
     } 
     while (((ret != -1) || (errno != ECHILD)) && !(timed_out = (Clock::now() > time_out)))  ;
-   
+#endif
+
     if (errno != ECHILD)
     {
         ec = boost::process::detail::get_last_error();
